pkg/mount-manager/safe-mounter_windows.go

```diff
@@ -0,0 +1,221 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mountmanager
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	diskv1alpha1 "github.com/kubernetes-csi/csi-proxy/client/api/disk/v1alpha1"
+	diskclientv1alpha1 "github.com/kubernetes-csi/csi-proxy/client/groups/disk/v1alpha1"
+
+	fsv1alpha1 "github.com/kubernetes-csi/csi-proxy/client/api/filesystem/v1alpha1"
+	fsclientv1alpha1 "github.com/kubernetes-csi/csi-proxy/client/groups/filesystem/v1alpha1"
+
+	volumev1alpha1 "github.com/kubernetes-csi/csi-proxy/client/api/volume/v1alpha1"
+	volumeclientv1alpha1 "github.com/kubernetes-csi/csi-proxy/client/groups/volume/v1alpha1"
+
+	utilexec "k8s.io/utils/exec"
+	"k8s.io/utils/mount"
+)
+
+var _ mount.Interface = &CSIProxyMounter{}
+
+type CSIProxyMounter struct {
+	FsClient     *fsclientv1alpha1.Client
+	DiskClient   *diskclientv1alpha1.Client
+	VolumeClient *volumeclientv1alpha1.Client
+}
+
+func NewCSIProxyMounter() (*CSIProxyMounter, error) {
+	fsClient, err := fsclientv1alpha1.NewClient()
+	if err != nil {
+		return nil, err
+	}
+	diskClient, err := diskclientv1alpha1.NewClient()
+	if err != nil {
+		return nil, err
+	}
+	volumeClient, err := volumeclientv1alpha1.NewClient()
+	if err != nil {
+		return nil, err
+	}
+	return &CSIProxyMounter{
+		FsClient:     fsClient,
+		DiskClient:   diskClient,
+		VolumeClient: volumeClient,
+	}, nil
+}
+
+func NewSafeMounter() (*mount.SafeFormatAndMount, error) {
+	csiProxyMounter, err := NewCSIProxyMounter()
+	if err != nil {
+		return nil, err
+	}
+	return &mount.SafeFormatAndMount{
+		Interface: csiProxyMounter,
+		Exec:      utilexec.New(),
+	}, nil
+}
+
+func (mounter *CSIProxyMounter) Mount(source string, target string, fstype string, options []string) error {
+	// Mount is called after the format is done.
+	// TODO: Confirm that fstype is empty.
+	// Call the LinkPath CSI proxy from the source path to the target path
+	parentDir := filepath.Dir(target)
+	if _, err := os.Stat(target); !os.IsNotExist(err) {
+		os.RemoveAll(target)
+	}
+	if err := os.MkdirAll(parentDir, 0755); err != nil {
+		return err
+	}
+	linkRequest := &fsv1alpha1.LinkPathRequest{
+		SourcePath: normalizeWindowsPath(source),
+		TargetPath: normalizeWindowsPath(target),
+	}
+	response, err := mounter.FsClient.LinkPath(context.Background(), linkRequest)
+	if err != nil {
+		return err
+	}
+	if response.Error != "" {
+		return errors.New(response.Error)
+	}
+	return nil
+}
+
+func (mounter *CSIProxyMounter) Unmount(target string) error {
+	rmdirRequest := &fsv1alpha1.RmdirRequest{
+		Path:    normalizeWindowsPath(target),
+		Context: fsv1alpha1.PathContext_POD,
+		Force:   true,
+	}
+	_, err := mounter.FsClient.Rmdir(context.Background(), rmdirRequest)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func normalizeWindowsPath(path string) string {
```

Comment:
Can we add some unit tests for this?

Reply:
actually mount utility already has the exact function and has unit test for it. I replace it with mount utitliy's one.